corehq/apps/users/tests/test_device_rate_limiter.py

```diff
@@ -1,77 +1,100 @@
-from django.test import SimpleTestCase, override_settings
+from django.test import TestCase
 from freezegun import freeze_time
 
 from corehq.apps.cloudcare.const import DEVICE_ID as CLOUDCARE_DEVICE_ID
-from corehq.apps.users.device_rate_limiter import device_rate_limiter
+from corehq.apps.users.device_rate_limiter import DEVICE_LIMIT_PER_USER_KEY, device_rate_limiter
+from corehq.project_limits.models import SystemLimit
 from corehq.tests.pytest_plugins.reusedb import clear_redis
-from corehq.util.test_utils import flag_enabled
+from corehq.util.test_utils import flag_disabled, flag_enabled
 
 
 @freeze_time("2024-12-10 12:05:43")
-@override_settings(DEVICE_LIMIT_PER_USER=1)
-@override_settings(ENABLE_DEVICE_RATE_LIMITER=True)
-class TestDeviceRateLimiter(SimpleTestCase):
+class TestDeviceRateLimiter(TestCase):
 
     domain = 'device-rate-limit-test'
 
     def setUp(self):
+        SystemLimit.objects.create(key=DEVICE_LIMIT_PER_USER_KEY, limit=1)
         self.addCleanup(clear_redis)
 
+    @flag_enabled("DEVICE_RATE_LIMITER")
     def test_allowed_if_no_devices_have_been_used_yet(self):
         self.assertFalse(device_rate_limiter.rate_limit_device(self.domain, 'user-id', 'new-device-id'))
 
-    @override_settings(DEVICE_LIMIT_PER_USER=2)
+    @flag_enabled("DEVICE_RATE_LIMITER")
```

Comment:
Can you do this on the class itself?

Reply:
it works at the class level, but then @flag_disabled didn't work for the one test where I wanted to use it since the patch had already been applied for the @flag_enabled decorator. I briefly looked to see if it would be an easy fix but came up empty handed and decided to just do this instead.

Reply:
that's annoying

Reply:
2d5aea65bd88ca69793c69f27ce2f12afc3334e6 (thank you @millerdev)